File: backend/app/presentation/schemas/resumes.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.application.resume_service import ResumeLevel


class ResumeNode(BaseModel):
    name: str
    path: str
    level: ResumeLevel | None = None
    child_count: int = Field(ge=0)
    modified_at: datetime
    status_id: str | None = None
    website_url: str | None = None


class ResumeTreeResponse(BaseModel):
    path: str
    level: ResumeLevel | None = None
    items: list[ResumeNode]


class ResumeNodeCreateRequest(BaseModel):
    path: str = ""
    name: str
    status_id: str | None = None


class ResumeNodeRenameRequest(BaseModel):
    path: str
    new_name: str


class VacancyFieldItem(BaseModel):
    name: str
    value: str = ""


class VacancyMeta(BaseModel):
    path: str
    name: str
    website_url: str = ""
    status_id: str | None = None
    fields: list[VacancyFieldItem]


class VacancyMetaPutRequest(BaseModel):
    website_url: str = ""
    status_id: str | None = None
    fields: list[VacancyFieldItem] = Field(default_factory=list)


class VacancyListItem(BaseModel):
    country: str
    company: str
    name: str
    path: str
    status_id: str | None = None
    website_url: str = ""
    modified_at: datetime


class VacancyListResponse(BaseModel):
    items: list[VacancyListItem]


class ResumeStatus(BaseModel):
    id: str | None = None
    name: str
    color: str


class ResumeStatusListResponse(BaseModel):
    statuses: list[ResumeStatus]


class ResumeStatusesPutRequest(BaseModel):
    statuses: list[ResumeStatus]

File: frontend/src/types/resumes.ts
export type ResumeNodeLevel = "COUNTRY" | "COMPANY" | "VACANCY";

export interface ResumeNode {
  name: string;
  path: string;
  level: ResumeNodeLevel | null;
  child_count: number;
  modified_at: string;
  status_id?: string | null;
  website_url?: string;
}

export interface ResumeTreeResponse {
  path: string;
  level: ResumeNodeLevel | null;
  items: ResumeNode[];
}

export interface ResumeField {
  name: string;
  value: string;
}

export interface VacancyMeta {
  path: string;
  name: string;
  website_url: string;
  status_id: string | null;
  fields: ResumeField[];
}

export interface VacancyMetaUpdate {
  website_url: string;
  status_id: string | null;
  fields: ResumeField[];
}

export interface ResumeStatus {
  id: string;
  name: string;
  color: string;
}

export interface ResumeStatusInput {
  id?: string;
  name: string;
  color: string;
}

export interface VacancyListItem {
  country: string;
  company: string;
  name: string;
  path: string;
  status_id: string | null;
  website_url: string;
  modified_at: string;
}

export interface VacancyListResponse {
  items: VacancyListItem[];
}

export interface ResumeStatusesResponse {
  statuses: ResumeStatus[];
}
